enterprise_subsidy/apps/api/v1/exceptions.py

```diff
@@ -0,0 +1,21 @@
+"""
+Custom exceptions for the Subsidy API
+"""
+from rest_framework.exceptions import APIException
+
+
+class ServerError(APIException):
+    """
+    A custom exception class for server errors.
+    This class is a subclass of the DRF APIException class, and is designed
+    to be raised when an unexpected error occurs in the subsidies service.
+    """
+    status_code = 500
+    default_detail = 'encountered an unexpected error in subsidies service'
+    default_code = 'server_error'
+
+    def __init__(self, *args, **kwargs):
+        self.code = kwargs['code']
+        self.developer_message = kwargs.pop('developer_message', None)
+        self.user_message = kwargs.pop('user_message', None)
+        super().__init__(*args, **kwargs)
```

Comment:
As discussed, I think I'd be interested in a way to have this result in a JSON response body including all the custom attributes when raised within a view.

Reply:
yeah, me too!  I think we can push this change through and follow up with a small PR focused on that aspect of things.

Reply:
Sounds like a plan, I'll follow up in a small PR!